Library/Homebrew/formulary.rb

```diff
@@ -251,4 +253,19 @@ def self.tap_paths(name)
       ]).detect(&:file?)
     end.compact
   end
+
+  def self.find_with_priority(ref, spec=:stable)
+    possible_pinned_tap_formulae = tap_paths(ref, Dir["#{HOMEBREW_LIBRARY}/PinnedTaps/*/*/"]).map(&:realpath)
+    if possible_pinned_tap_formulae.size > 1
+      raise TapFormulaAmbiguityError.new(ref, possible_pinned_tap_formulae)
+    elsif possible_pinned_tap_formulae.size == 1
+      selected_formula = factory(possible_pinned_tap_formulae.first, spec)
+      if core_path(ref).file?
+        opoo "#{ref} is provided by core, but is now shadowed by #{selected_formula.tap}."
```

Comment:
`selected_formula.tap` -> `selected_formula.full_name`.

Also maybe add `"To refer to the core formula, use Homebrew/homebrew/#{ref}"`.